Simplify Bazel

cpsauer opened the pull request:
Please see #309 for context, but this sidesteps the bazel dependencies and cleans up the copts where they're not actually used.

Fixes #309

zaucy commented on test/BUILD.bazel:
```diff
@@ -1,23 +1,25 @@
+_COPTS = select({
+    "//conditions:default": ["-std=c++17"],
+    "@bazel_tools//tools/cpp:clang-cl": _MSVC_FLAGS,
+    "@bazel_tools//tools/cpp:msvc": _MSVC_FLAGS, 
```
These options don't work unless specifying platforms as far as I know. Example from my previous bzlmod PR https://github.com/Neargye/magic_enum/pull/254/files#diff-e13b51fb7feeb38d3fb7dee88724107c57fb3d4805068b5f8d57a2d0900fb0dc. It'll just fallback to the default.

e.g.
```
cl : Command line warning D9002 : ignoring unknown option '-std=c++17'
```
Followed by many errors.

cpsauer replied:
Ah shoot. Sorry I missed that. The other one worked without platforms, I take it? Presumably the toolchains set the @bazel_tools//tools/cpp:compiler flag internally?

I think the right move, then, is to re-add the bazelmod rules_cc dependency for tests (only). Seems cleaner than duplicating the config settings and still accomplishes all the previous goals around consuming the library itself. I'll push up a commit that does just that.

zaucy reviewed (changes requested):
If there's a way to not declare platforms in the test (and everyone upstream who uses bzlmod) that would be great.

cpsauer committed:
Simplify Bazel
Please see #309 for context, but this sidesteps the bazel dependencies and cleans up the copts where they're not actually used.

Neargye committed:
Merge branch 'master' into no-dep-on-rules-cc

zaucy commented on test/MODULE.bazel:
```diff
@@ -1,6 +1,6 @@
-
 local_path_override(module_name = "magic_enum", path = "..")
+
+bazel_dep(name = "rules_cc", version = "0.0.8")
```
nit: maybe put the bazel dep back up top? :P

cpsauer replied:
I'd reordered intentionally, thinking we might want the magic-enum dependency configuration all together for clarity that it's being overridden and quick-skimmability.

But I don't feel super strongly or anything--feel free to change if you feel strongly the other way? Or generally lmk if you think there's a different principle I should be absorbing here.

(I'm assuming from the approval that you don't want to block merging on this, yeah? Still v interested in what you have to say, ofc.)

zaucy reviewed (approved):
Tested on one of my projects and it seems fine to me! :+1: (although I only tested on Windows)

cpsauer commented:
Thanks so much for testing @zaucy, and working so thoughtfully with me to make things even better. I really appreciate your being great to collaborate with!

All good to merge, I think, yeah?

@Neargye, thanks to you too for being so responsive and great!

Cheers,
Chris

Neargye commented:
@cpsauer  @zaucy Thanks for help!

cpsauer commented:
Thank *you*, Daniil!

cpsauer commented:
@Neargye, how imminent is v0.9.5, do you think? Trying to decide whether to switch to commit-by-commit to get things back working w/ Bazel vs backpin/waiting for the next release.

Neargye commented:
@cpsauer v0.9.5 is released

cpsauer commented:
Yay! Thanks :)
